Sources/_Core.swift

```diff
@@ -43,79 +42,16 @@ fileprivate let NamedCaptureGroupsPattern = try! NSRegularExpression(
   , options: .dotMatchesLineSeparators
   )
 
-fileprivate typealias _GroupNamesSearchResult = (
-    _outerOrdinaryCaptureGroup: NSTextCheckingResult
-  , _innerRefinedNamedCaptureGroup: NSTextCheckingResult
-  , _index: Int
-  )
-
-/// This class is a by-product of tradeoff between interoperatablitiy 
-/// of Swift and ObjC, and Swift's sophisticated access control system.
-///
-/// It is an implementation detail of the lib's core and is subject to
-/// future changes; do not use it directly.
-public final class _ObjCGroupNamesSearchResult: NSObject {
-  public let _outerOrdinaryCaptureGroup: NSTextCheckingResult
-  public let _innerRefinedNamedCaptureGroup: NSTextCheckingResult
-  public let _index: Int
-
-  @nonobjc fileprivate init( _ tuple: _GroupNamesSearchResult ) {
-    _outerOrdinaryCaptureGroup = tuple._outerOrdinaryCaptureGroup
-    _innerRefinedNamedCaptureGroup = tuple._innerRefinedNamedCaptureGroup
-    _index = tuple._index
-    }
-  }
 
 public extension NSRegularExpression /* _NamedCaptureGroupsSupport */ {
```

Comment:
Without having to do ObjC *mix and match*, this extension should be made `internal`. Just dropping the `public` is okay.